rspace/src/main/scala/coop/rchain/rspace/trace/Event.scala

```diff
@@ -34,23 +34,29 @@ object Event {
   implicit def codecLog: Codec[Seq[Event]] = codecSeq[Event](codecEvent)
 }
 
-case class COMM(consume: Consume, produces: Seq[Produce]) extends Event
+case class COMM(consume: Consume, produces: Seq[Produce]) extends Event {
+  def nextSequenceNumber: Int =
+    Math.max(
+      consume.sequenceNumber,
+      produces.map(_.sequenceNumber).max
+    ) + 1
+}
 
 object COMM {
-
   implicit val codecCOMM: Codec[COMM] = (Codec[Consume] :: Codec[Seq[Produce]]).as[COMM]
 }
 
 sealed trait IOEvent extends Event
 
-case class Produce private (channelsHash: Blake2b256Hash, hash: Blake2b256Hash) extends IOEvent {
+case class Produce private (channelsHash: Blake2b256Hash, hash: Blake2b256Hash, sequenceNumber: Int)
+    extends IOEvent {
 
   override def equals(obj: scala.Any): Boolean = obj match {
-    case produce: Produce => produce.hash == hash
+    case produce: Produce => produce.hash == hash && produce.sequenceNumber == sequenceNumber
```

Comment:
The hash includes it already. channelsHash was so that the Produce's channel could be identified even if the datum/persist flag differed - turns out it would have been useful to have all three channel, datum, and persist be intact for debugging purposes.